### ISSUE TITLE
Destructuring not recognized

### ISSUE THREAD

**Issue by henry2004y** [JuliaSIMD/Polyester.jl#75]
Hi,

I've encountered a case where Polyester.jl does not know how to handle the destructuring syntax introduced in Julia 1.7:

```julia
using Polyester

struct A
   a
   b
end

x = A(1,2)

Threads.@threads for i in 1:2
   (; a, b) = x
end

@batch for i in 1:2
   (; a, b) = x
end
```

The `@batch` loop returns
```
ERROR: LoadError: "Don't know how to handle:\n \$(Expr(:parameters, :a, :b))"
```

**Comment by Moelf** [comment]
bump, do we need a pass in the macro to transform this into a bunch of getproperty?

**Comment by chriselrod** [comment]
> bump, do we need a pass in the macro to transform this into a bunch of getproperty?

Yeah, that'd be a reasonable approach. Want to take a stab at it?

**Comment by Moelf** [comment]
where even is this signature defined
>       define_tup!(defined, a)

**Comment by Moelf** [comment]
also this doesn't quite work, this logic assumes each element in `ex.args` gets replaced by a gensym

but when `a` is a `:parameters` expression, it inflates into multiple `ex.args`
